content/blog/secure-cloud-kubernetes-workloads-intro/index.md

```diff
@@ -0,0 +1,99 @@
+
+### AWS EKS
+
+EKS initially required the install of third-party solutions like [KIAM](https://github.com/uswitch/kiam) and [Kube2IAM](https://github.com/jtblin/kube2iam) in the cluster -- both of which are solutions that are widely used and well tested.
+
+EKS now offers a native solution, which involves creating a cluster with an OpenID connect issuer attached to it. You can read more about this [here](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html).
+
+Once you've created a cluster and associated an IAM OIDC provider to it, you can deploy Kubernetes pods which use an [AWS IAM Role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html) which you'll need to create. To leverage this IAM role in your pod, you have to patch the Kubernetes [service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) associated with the pod and add annotations that tell the pod which IAM role to use.
+
+Pulumi's first class support for Kubernetes Helm Charts and AWS resources mean you can perform all of this in one Pulumi program. Look out for the rest of this series to see how it's done!
+
+### Azure AKS
+
+Azure AKS uses a different approach to assign roles to pods. [AAD Pod Identity](https://github.com/Azure/aad-pod-identity) project is an add-on deployed into your AKS cluster. AAD Pod Identity includes a Kubernetes [CustomResource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/), which the running components of AAD Pod Identity watch. You can then create a [Service Principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals) and assign it to your pod.
+
+Again, Pulumi has first class support for all of Azure's resource via the [Azure NextGen Provider](https://github.com/pulumi/pulumi-azure-nextgen). Watch out for the rest of the series for a real world example!
+
+### Google Cloud GKE
+
+Google's GKE has native support for what it calls ["Workload Identity"](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity). With this solution, you create a GKE cluster with workload identity enabled, and ensure that at least one node pool has workload identity enabled. You can then annotate the Kubernetes service account associated with your pod and point it at a created [GCP Service Account](https://cloud.google.com/iam/docs/service-accounts). Hopefully, the confusion between Kubernetes service accounts and Google Cloud service accounts will become clearer in the tutorial blog post!
+
+## Wrap up
+
+We've detailed compelling case for why you should reconsider using static credentials for your application, and introduced the idea for how to avoid using them in your cloud-native applications. In our next post, I'll introduce the application we're going to deploy to all of these different cloud providers, and show you how to deploy it securely on AWS EKS. Stay tuned!
```

Comment:
```suggestion
We've detailed a compelling case for why you should reconsider using static credentials for your application and introduced how to avoid using them in your cloud-native applications. In our next post, I'll introduce the application we're going to deploy to all of the different cloud providers and show you how to deploy it securely on AWS EKS. Stay tuned!
```